test/fitnesse/junit/JUnitXMLPerPageRunListenerIntegrationTest.java

```diff
@@ -0,0 +1,103 @@
+package fitnesse.junit;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.Description;
+import org.junit.runner.notification.Failure;
+
+import java.io.IOException;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+public class JUnitXMLPerPageRunListenerIntegrationTest {
```

Comment:
Given you have tested the xml writer separately I would suggest we keep only the the IntegrationTest or test with mocks.

Reply:
Deleted integration test, kept unit test.